spiped/dispatch.c

```diff
@@ -94,13 +103,34 @@ doaccept(struct accept_state * A)
 
 /* A connection has closed.  Accept more if necessary. */
 static int
-callback_conndied(void * cookie)
+callback_conndied(void * cookie, void * conn_cookie)
 {
 	struct accept_state * A = cookie;
+	struct conn_list_node * node_ptr;
+
+	assert(A->conn_cookies != NULL);
 
 	/* We've lost a connection. */
 	A->nconn -= 1;
 
+	/* Remove conn_cookie from the list of conn_cookies. */
+	node_ptr = A->conn_cookies;
+	while (node_ptr->conn_cookie != conn_cookie)
```

Comment:
Why are you walking the list?  Why not pass the node pointer as the cookie to be returned to us?